### PR TITLE
Create KMP.c

### PR DESCRIPTION
**Pull request by Harikrishnan6336**
Fixes #2555 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Added a file KMP.c
- 

#### Languages Used:
- C
- 

#### Files Added:
- KMP.c
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Knuth_Morris_Pratt_Algorithm/KMP.c [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+int* makeTable(char* pattern) {
+
+	int patternSize = strlen(pattern);
+	int *table = (int*)malloc(patternSize*sizeof(int));
+	int j = 0; int i = 1 ;
+	memset(table, 0, sizeof(table));
+	for (i = 1; i <= patternSize; i++) {
+		while (j > 0 && pattern[i] != pattern[j]) {
+			j = table[j - 1];    // If there is a mismatch, move j one by one until the last match. At this time, j = table [j-1].
```
Comment (asha15) [inline Knuth_Morris_Pratt_Algorithm/KMP.c:18]: take this comment to the previous line

File: Knuth_Morris_Pratt_Algorithm/KMP.c [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+		if (pattern[i] == pattern[j]) {
+			table[i] = ++j; 
+		}
+	}
+	return table;
+}
+
+void KMP_ALGORITHM(char* originaltext, char*pattern) {
+
+	int patternlen = strlen(pattern);
```
Comment (raghav-dalmia) [inline Knuth_Morris_Pratt_Algorithm/KMP.c:29]: follow 4 space indentation

File: Knuth_Morris_Pratt_Algorithm/KMP.c [hunk 3]
```diff
@@ -0,0 +1,90 @@
+//Implementation of the Knuth Morris Pratt algorithm for pattern searching in C language
+#include<stdio.h>
+#include<stdlib.h>
+#include<string.h>
+#include<memory.h>
+
+char *parent = "acabacdabac";
```
Comment (raghav-dalmia) [inline Knuth_Morris_Pratt_Algorithm/KMP.c:7]: Remove these variables(you haven't used them) and if required, input them dynamically.

File: Knuth_Morris_Pratt_Algorithm/KMP.c [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+			j = table[j - 1];   
+		}
+		if (pattern[i] == pattern[j]) {
+			table[i] = ++j; 
+		}
+	}
+	return table;
+}
+
+void KMP_ALGORITHM(char* originaltext, char*pattern) {
```
Comment (raghav-dalmia) [inline Knuth_Morris_Pratt_Algorithm/KMP.c:27]: add space around *

File: Knuth_Morris_Pratt_Algorithm/KMP.c [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+char *pattern = "abacdab";
+
+int* makeTable(char* pattern) {
+
+	int patternSize = strlen(pattern);
+	int *table = (int*)malloc(patternSize*sizeof(int));
+	int j = 0; int i = 1 ;
+	memset(table, 0, sizeof(table));
+	for (i = 1; i <= patternSize; i++) {
+		while (j > 0 && pattern[i] != pattern[j]) { // If there is a mismatch, move j one by one until the last match. At this time, j = table [j-1].
```
Comment (raghav-dalmia) [inline Knuth_Morris_Pratt_Algorithm/KMP.c:17]: one line can have at max 80 characters

**Comment by raghav-dalmia** [comment]
Also, squash your commits.

**Comment by somya-kapoor** [comment]
Are you working on it?

**Comment by somya-kapoor** [comment]
Due to no activity for more than 7 days, I am closing this issue.

**Pull request closed without merging**